src/main/java/io/debezium/connector/yugabytedb/YugabyteDBEventDispatcher.java

```diff
@@ -20,37 +30,36 @@
 import io.debezium.util.SchemaNameAdjuster;
 
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.Optional;
+
 /**
  * Custom extension of the {@link EventDispatcher} to accommodate routing {@link LogicalDecodingMessage} events to the change event queue.
  *
- * @author Lairen Hightower
+ * @author Rajat Venkatesh, Vaibhav Kushwaha
  */
 public class YugabyteDBEventDispatcher<T extends DataCollectionId> extends EventDispatcher<YBPartition, T> {
     private static final Logger LOGGER = LoggerFactory.getLogger(YugabyteDBEventDispatcher.class);
+    private final YugabyteDBConnectorConfig connectorConfig;
+    private final ChangeEventCreator changeEventCreator;
     private final ChangeEventQueue<DataChangeEvent> queue;
+    private final boolean emitTombstonesOnDelete;
     private final LogicalDecodingMessageMonitor logicalDecodingMessageMonitor;
     private final LogicalDecodingMessageFilter messageFilter;
-
-    public YugabyteDBEventDispatcher(YugabyteDBConnectorConfig connectorConfig, TopicSelector<T> topicSelector,
-                                   DatabaseSchema<T> schema, ChangeEventQueue<DataChangeEvent> queue, DataCollectionFilters.DataCollectionFilter<T> filter,
-                                   ChangeEventCreator changeEventCreator, EventMetadataProvider metadataProvider, SchemaNameAdjuster schemaNameAdjuster) {
-        this(connectorConfig, topicSelector, schema, queue, filter, changeEventCreator, null, metadataProvider,
-                new HeartbeatFactory<>(connectorConfig, topicSelector, schemaNameAdjuster), schemaNameAdjuster, null);
-    }
-
-    public YugabyteDBEventDispatcher(YugabyteDBConnectorConfig connectorConfig, TopicSelector<T> topicSelector,
-                                   DatabaseSchema<T> schema, ChangeEventQueue<DataChangeEvent> queue, DataCollectionFilters.DataCollectionFilter<T> filter,
-                                   ChangeEventCreator changeEventCreator, EventMetadataProvider metadataProvider,
-                                     HeartbeatFactory<T> heartbeatFactory, SchemaNameAdjuster schemaNameAdjuster) {
-        this(connectorConfig, topicSelector, schema, queue, filter, changeEventCreator, null, metadataProvider,
-                heartbeatFactory, schemaNameAdjuster, null);
-    }
+    private final TopicSelector<T> topicSelector;
+    private final DatabaseSchema<T> schema;
+    private DataChangeEventListener<YBPartition> eventListener = DataChangeEventListener.NO_OP();
+    private final InconsistentSchemaHandler<YBPartition, T> inconsistentSchemaHandler;
+    private final Signal<YBPartition> signal;
+    private final boolean neverSkip;
+    private final Heartbeat heartbeat;
+    private final EnumSet<Envelope.Operation> skippedOperations;
+    private final DataCollectionFilters.DataCollectionFilter<T> filter;
+    private final YugabyteDBTransactionMonitor transactionMonitor;
+    private final YugabyteDBStreamingChangeRecordReceiver streamingReceiver;
```

Comment:
Changes part of implementing custom transaction monitor to publish transaction metadata.